v2/mongodb-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/mongodb/templates/MongoDbUtils.java

```diff
@@ -76,4 +84,56 @@ public static Document getMongoDbDocument(String uri, String dbName, String coll
     Document doc = collection.find().first();
     return doc;
   }
+
+  public static TableRow getTableSchemaCDC(HashMap<String, Object> parsedMap, String userOption) {
+    TableRow row = new TableRow();
+    if (userOption.equals("FLATTEN")) {
+      parsedMap.forEach(
+          (key, value) -> {
+            String valueClass = value.getClass().getName();
+            switch (valueClass) {
+              case "java.lang.Double":
+                row.set(key, value);
+                break;
+              case "java.util.Integer":
+                row.set(key, value);
+                break;
+              default:
+                row.set(key, value.toString());
+            }
+          });
+    } else {
+      LocalDateTime localdate = LocalDateTime.now(ZoneId.of("UTC"));
+      row.set("id", parsedMap.get("_id").toString())
+          .set("source_data", parsedMap.toString())
+          .set("timestamp", localdate.format(TIMEFORMAT));
+    }
+    return row;
+  }
+
+  public static TableRow getTableSchema(Document document, String userOption) {
```

Comment:
This method seems to be doing the same thing as `getTableSchemaCDC(HashMap<String, Object> parsedMap, String userOption)`, and `Document` implements `Map<String, Object>`, so a document can be passed to the `getTableSchemaCDC` method too.

Maybe remove this method and use the one that accepts a Map? And then rename it from getTableSchemaCDC to getTableSchema or something more generic.

Reply:
Updated. Removed the method getTableSchema and renamed getTableSchemaCDC to getTableSchema.